File: schemas/booking-scheduling/drizzle/booking_custom_field_answers.ts
// booking_custom_field_answers: customer responses to custom intake fields for a booking.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index, unique } from "drizzle-orm/pg-core";
import { bookings } from "./bookings";
import { customFields } from "./custom_fields";

export const bookingCustomFieldAnswers = pgTable(
  "booking_custom_field_answers",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    bookingId: uuid("booking_id").notNull().references(() => bookings.id, { onDelete: "cascade" }),
    customFieldId: uuid("custom_field_id").notNull().references(() => customFields.id, { onDelete: "cascade" }),
    value: text("value").notNull(),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    unique("uq_booking_custom_field_answers_booking_id_custom_field_id").on(table.bookingId, table.customFieldId),
    index("idx_booking_custom_field_answers_custom_field_id").on(table.customFieldId),
  ]
);

File: schemas/booking-scheduling/spacetimedb/booking_custom_field_answers.rs
// booking_custom_field_answers: Customer responses to custom intake fields for a booking.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = booking_custom_field_answers, public)]
pub struct BookingCustomFieldAnswer {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub booking_id: String, // UUID — FK → bookings.id (cascade delete)
    #[index(btree)]
    pub custom_field_id: String, // UUID — FK → custom_fields.id (cascade delete)
    pub value: String,
    pub created_at: Timestamp,
    // Composite unique: (booking_id, custom_field_id)
}
